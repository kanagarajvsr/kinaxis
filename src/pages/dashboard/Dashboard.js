import Widget from "../../components/widgets/Widgets";
import Featured from "../../components/featured/Featured";
import Chart from "../../components/chart/Chart";
import Datatable from "../../components/datatable/Datatable";
import { Columns, Rows } from "../../jsonData/transactionData";
import { widgets, charts } from "../../jsonData/dashboardData";
import "./dashboard.scss"; 
import { useEffect, useState } from "react";


const Dashboard = () => {
  const [filter,setFilter] = useState(6);
  const [chartData,setChartData] = useState(6);

  useEffect(()=>{
    setChartData(charts.slice(0,filter));
  },[filter])
 
  return (
    <div  className="dashboard">
      <div className="widgets">
        <Widget data={ widgets.user} />
        <Widget data={ widgets.order} />
        <Widget data={ widgets.earning} />
        <Widget data={ widgets.balance} />
      </div>
      <div className="charts">
          <Featured />
          <div className="chart">
          <div className="title">Last {filter} Months (Revenue)  
            <select data-testid="select-id" className="select" value={filter} label="Age" onChange={(e)=>setFilter(e.target.value)} >
              <option value={3}>Last 3 Months (Revenue)</option>
              <option value={6}>Last 6 Months (Revenue)</option>
              <option value={12}>Last 12 Months (Revenue)</option>
            </select>
           </div>
          <Chart  data={chartData} aspect={2 / 1} />
          </div>
      </div>
      <div className="listContainer">
          <Datatable columns={Columns} rows={Rows} pageName="Latest Transactions" />
      </div>
    </div>
  )
}

export default Dashboard;