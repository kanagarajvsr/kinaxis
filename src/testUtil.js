
import { BrowserRouter } from 'react-router-dom'
import { render,screen,fireEvent } from '@testing-library/react';
 
export const renderWithRouter = (ui, {route = '/'} = {}) => {
    window.history.pushState({}, 'Test page', route);
    const renders = render(ui, {wrapper: BrowserRouter})
    return { 
        screen,
        fireEvent,
        ...renders
    }
  }