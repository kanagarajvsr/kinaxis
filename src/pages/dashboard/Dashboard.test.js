import Dashboard from './Dashboard';
import { cleanup } from '@testing-library/react';
import {renderWithRouter} from '../../testUtil';

describe('Dashboard render', () => {
  afterEach(cleanup);
  const utils =  renderWithRouter(<Dashboard />)
  const input = utils.getByTestId("select-id");
  const {asFragment,fireEvent}=utils;
   
  test('transaction onchange with 3 months', () => {
      fireEvent.change(input, {target: {value: '3'}})
      expect(input.value).toBe('3')
  })
 
  test('transaction onchange with 6 months', () => {
      fireEvent.change(input, {target: {value: '6'}})
      expect(input.value).toBe('6') 
  })

  test('Update Snapshot', () => { 
    expect(asFragment()).toMatchSnapshot()
  })

});
   


