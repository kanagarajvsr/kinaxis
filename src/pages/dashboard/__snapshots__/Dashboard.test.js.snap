// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard render Update Snapshot 1`] = `<DocumentFragment />`;
